rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -806,20 +840,118 @@ public License convertLicenseFromRequest(Map<String, Object> reqBodyMap, License
+     */
+    private void isLicenseValidResilient(Set<String> licenses, String fieldName) {
+        if (CommonUtils.isNullOrEmptyCollection(licenses)) {
+            return;
+        }
+
+        List<String> licenseIncorrect = new ArrayList<>();
+        List<String> licenseWarnings = new ArrayList<>();
+        
+        for (String licenseId : licenses) {
+            if (licenseId == null || licenseId.trim().isEmpty()) {
+                continue; // Skip empty/null license IDs
+            }
+            if (isLikelyExternalLicenseReference(licenseId)) {
+                licenseWarnings.add(licenseId);
+                LOGGER.info("{} ID '{}' appears to be external license reference - allowing without database validation", 
+                          fieldName, licenseId);
+                continue;
+            }
+
+            try {
+                licenseService.getLicenseById(licenseId);
+                LOGGER.debug("{} validation successful for ID: {}", fieldName, licenseId);
+            } catch (Exception e) {
+                licenseIncorrect.add(licenseId);
+                LOGGER.warn("{} ID '{}' not found in SW360 database and not recognized as external reference", 
+                          fieldName, licenseId);
             }
         }
+
         if (!licenseIncorrect.isEmpty()) {
-            throw new BadRequestClientException("License with ids " + licenseIncorrect + " does not exist in SW360 database.");
+            String errorMsg = fieldName + " with ids " + licenseIncorrect + " does not exist in SW360 database.";
+            if (!licenseWarnings.isEmpty()) {
+                errorMsg += " Note: External license references " + licenseWarnings + " were allowed.";
+            }
+            LOGGER.error("License validation failed: {}", errorMsg);
+            throw new BadRequestClientException(errorMsg);
         }
+
+        if (!licenseWarnings.isEmpty()) {
+            LOGGER.info("{} update proceeding with external license references: {}", fieldName, licenseWarnings);
+        }
+    }
+
+    /**
+     * Helper method to determine if a license ID appears to be an external reference
+     * (e.g., long descriptive names, proprietary licenses) rather than an internal SW360 ID
+     * This method is conservative to avoid false positives with common SPDX identifiers
+     */
+    private boolean isLikelyExternalLicenseReference(String licenseId) {
+        if (licenseId == null || licenseId.trim().isEmpty()) {
+            return false;
+        }
+
+        String normalizedId = licenseId.toLowerCase().trim();
+
+        return isMatchingConfigurablePatterns(normalizedId) ||
+               isLongDescriptiveName(normalizedId);
+    }
+
+    /**
+     * Check against configurable license patterns from application properties
+     * This is now more conservative to avoid false positives with standard SPDX identifiers
+     */
+    private boolean isMatchingConfigurablePatterns(String normalizedId) {
```

Comment:
This function does not make sense to me. Can you please explain why are we looking for Microsoft/Oracle/Google etc in the license names and calling them external licenses?